lib/matplotlib/artist.py

```diff
@@ -284,11 +286,28 @@ def get_tightbbox(self, renderer):
 
     def add_callback(self, func):
         """
-        Adds a callback function that will be called whenever one of
-        the :class:`Artist`'s properties changes.
+        Add a callback function that will be called whenever one of the
+        `.Artist`'s properties changes.
 
-        Returns an *id* that is useful for removing the callback with
-        :meth:`remove_callback` later.
+        Parameters
+        ----------
+        func : Callable[Artist]
```

Comment:
Technically that's `Callable[[Artist], Any]` (I think).

Reply:
Reverted to just `callback`. In such cases, I will write the required signature as a type-annotated example in such cases.

IMO `def func(artist: Artist) -> Any` is easier to understand than `Callable[[Artist], Any]`.